pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -150,13 +154,32 @@ def _build_osv_data(name, ecosystem, version=None, next_page_token=None):
     return osv_data
 
 
-def _identify_package_ecosystem(content: any) -> str:
+def _identify_package_ecosystem(content: any, repository=None) -> str:
     """
     Returns an osv.dev ecosystem (string) based on the content_type
     """
     if isinstance(content, NPMPackage):
-        return getattr(PKG_ECOSYSTEM, "npm", None)
+        return [getattr(PKG_ECOSYSTEM, "npm", None)]
     elif content.TYPE in ["python", "gem"]:
-        return getattr(PKG_ECOSYSTEM, content.TYPE, None)
+        return [getattr(PKG_ECOSYSTEM, content.TYPE, None)]
+    elif repository and repository.pulp_type == "rpm.rpm":
+        if content.pulp_type == "rpm.package":
+            return _convert_rhel_repo_cpe(repository)
+        else:
+            # ignore non rpm packages (advisory, packagecategory, packagelangpacks)
+            return []
     else:
         raise RuntimeError("Package type not supported!")
+
+
+def _convert_rhel_repo_cpe(repo):
+    """
+    Convert the CPE into osv.dev expected format
+    """
+    cpe_prefix = r"^cpe:\/[oa]:redhat"
```

Comment:
Since this function is called N times, it would be more efficient to move this outside of the function and make it `pattern = re.compile(r"your_regex_here")` then use it over and over.